src/filter/mod.rs

```diff
@@ -199,17 +199,58 @@ pub fn box_filter(image: &GrayImage, x_radius: u32, y_radius: u32) -> Image<Luma
     out
 }
 
+/// Returns 2d correlation of an image. Intermediate calculations are performed
+/// at type K, and the results converted to pixel Q via f. Pads by continuity.
+pub fn filter<P, K, F, Q>(image: &Image<P>, kernel: &Kernel<K>, mut f: F) -> Image<Q>
```

Comment:
I agree that making this a freestanding function increases discoverability and is more in keeping with the rest of the library.